krm/gitlab-krm-operator/README.md

```diff
@@ -0,0 +1,89 @@
+# Gitlab KRM Operator
+
+  Gitlab KRM Operator is a kubernetes extension that implements Kubernetes style APIs for Gitlab resources.
+  Currently it supports following APIs:
+
+  - Groups
+  - Projects
+  - DeployKeys
+
+# Prerequisites
+
+  Before you begin installing this operator in a kubernetes cluster, you will need the following:
+
+  - Kubernetes (v1.14+) cluster
+  - Kustomize
+  - Kubectl configured to talk to the kubernetes cluster
+  - Gitlab Base API URL
+  - Access Token to access the APIs
+
+# Installation
+
+  1. Edit the kustomization file to replace the Gitlab base API URL and Gitlab access token as shown below:
+
+  ```yaml
+namePrefix: gitlab-cnrm-
+namespace: gitlab-cnrm-system
+resources:
+  - ...
+secretGenerator:
+  - name: gitlab-creds
+    literals:
+    - gitlab-token="GITLAB_ACCESS_TOKEN" <--- Put Gitlab access token here
+configMapGenerator:
+  - name: gitlab-config
+    literals:
+      - gitlab-base-url="GITLAB BASE API URL" <-- Put Gitlab API URL
```

Comment:
Suggest adding comments in the yaml file for these fields that need users' input